example_cpp_smart_card_client_app/src/window.js

```diff
@@ -47,11 +56,45 @@ function onCloseWindowClicked(e) {
   chrome.app.window.current().close();
 }
 
-// Set up event listeners.
+// Called when the "run test" button is clicked. Sends a command to the NaCl
+// module to start the test.
+function onRunTestClicked(e) {
+  e.preventDefault();
+  naclModuleMessageChannel.send('ui_backend', {command: 'run_test'});
+}
+
+// Called when a "output_message" message is received from the NaCl module. Adds
+// a line into the "output" <pre> element in the UI.
+function displayOutputMessage(text) {
+  const outputElem = goog.dom.getElement('output');
+  outputElem.innerHTML = text + '\n' + outputElem.innerHTML;
```

Comment:
Without seeing how it looks in practice, based on various chat examples that come to my mind, I would expect that messages should be appended and not prepended. Of course, this only works if there is scrolling etc. when there are too many messages to fit the window.

Reply:
Done (there's no scrolling, but we don't care about it too much, as it's just an example without real styling).